### DISCUSSION TITLE
RPC Error code 500 with setProviderByName

### DISCUSSION THREAD

**Discussion by Michele971** [reach-sh/reach-lang#1411] [Q&A] [answered]
Hi guys,
the RPC command `rpc('/stdlib/setProviderByName','TestNet')` gave me this **500 code error**:
`"requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://127.0.0.1:3000/stdlib/setProviderByName"`.

What can I do?

**Comment by jeapostrophe** [comment, marked as answer]
There should be an error in the server log

**Reply by Michele971** [reply]
So... What do I have to do ? This is my fault?

**Reply by jeapostrophe** [reply]
We can't know until you look at the log and see what the error is

**Reply by Michele971** [reply]
The log messages is: 
`urllib3/util/retry.py", line 573, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='127.0.0.1', port=3000): Max retries exceeded with url: /stdlib/setProviderByName (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x107d0bcc0>: Failed to establish a new connection: [Errno 61] Connection refused'))`.

It is weird, because every other RPC call is fine. This is the only one that fails.

**Reply by jeapostrophe** [reply]
That error looks like a client error... there might be a different error in the server log?

**Reply by Michele971** [reply]
First of all, thanks for your help. 
No, this is the only error that I have in the server log.
I tried removing the containers and install them again, but nothing happened. Same error.

**Reply by Michele971** [reply]
I found a new log:
```
2022-09-13T13:19:40.238Z DEBUG: RPC /stdlib/setProviderByName TestNet
2022-09-13T13:19:40.239Z DEBUG: !! Witnessed exception triggered by client ::ffff:172.21.0.1: POST /stdlib/setProviderByName TestNet:
  Error: setProviderEnv called after it was already set
    at setProviderEnv (/stdlib/dist/cjs/ETH_impl.js:309:15)
    at setProviderByEnv (/stdlib/dist/cjs/ETH_impl.js:256:5)
    at Object.setProviderByName (/stdlib/dist/cjs/ETH_impl.js:261:5)
    at /stdlib/dist/cjs/rpc_server.js:514:69
    at step (/stdlib/dist/cjs/rpc_server.js:44:23)
    at Object.next (/stdlib/dist/cjs/rpc_server.js:25:53)
    at /stdlib/dist/cjs/rpc_server.js:19:71
    at new Promise (<anonymous>)
    at __awaiter (/stdlib/dist/cjs/rpc_server.js:15:12)
    at /stdlib/dist/cjs/rpc_server.js:505:88
2022-09-13T13:19:40.241Z DEBUG: !! HTTP 500: "Unspecified fault" response sent to client
Traceback (most recent call last):
```

I did not set setProviderEnv ...

**Reply by jeapostrophe** [reply]
Did you call any other functions like it? Maybe you could give a gist of the whole log?

**Reply by Michele971** [reply]
I don't call any other functions like it. Anyways, I just re-started (again) everything and now it works... Thanks..
